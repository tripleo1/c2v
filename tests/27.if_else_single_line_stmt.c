#include <stdio.h>

int main() {
	int x = 1;

	// else while
	if (x == 2) {
		printf("x=2\n");
	} else while (x!=2) {
		x=2;
	}

	// else goto
	if (x == 2) {
		printf("x=2\n");
	} else goto done;

	// else switch
	if (x == 2) {
		printf("x=2\n");
		} else switch (x) {
			case 1 :
				break;
			default:
				break;
		}

	// else do
	if (x == 2) {
		printf("x=2\n");
		} else do {
		} while(0);

	// else for
	if (x == 2) {
		printf("x=2\n");
	} else for (int i=0; i<2; i++) {
		printf("x!=2\n");
	}

done:
	return 100;
}
