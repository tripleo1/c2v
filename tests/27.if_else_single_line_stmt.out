@[translated]
module main

fn main() {
	x := 1
	// else while
	if x == 2 {
		C.printf(c'x=2\n')
	} else {
		for x != 2 {
			x = 2
		}
	}
	// else goto
	if x == 2 {
		C.printf(c'x=2\n')
	} else {
		unsafe {
			goto done
		}
	}
	// else switch
	if x == 2 {
		C.printf(c'x=2\n')
	} else {
		match x {
			1 { // case comp body kind=BreakStmt is_enum=false
			}
			else {}
		}
	}
	// else do
	if x == 2 {
		C.printf(c'x=2\n')
	} else {
		for {
			// while()
			if !(0) {
				break
			}
		}
	}
	// else for
	if x == 2 {
		C.printf(c'x=2\n')
	} else {
		for i := 0; i < 2; i++ {
			C.printf(c'x!=2\n')
		}
	}
	done:
	return
}
